web/htdocs/assets/js/codesearch_ui.js

```diff
@@ -45,10 +47,89 @@ function shorten(ref) {
   return ref;
 }
 
-var Match = Backbone.Model.extend({
+var MatchView = Backbone.View.extend({
+  tagName: 'div',
   initialize: function() {
+    this.model.on('change', this.render, this);
   },
-  url: function() {
+  render: function() {
+    var div = this._render();
+    this.setElement(div);
+    return this;
+  },
+  _renderLno: function(n, isMatch) {
+    var lnoStr = n.toString() + (isMatch ? ":" : "-");
+    var classes = ['lno'];
+    if (isMatch) classes.push('matchlno');
+    return h.a({cls: 'lno-link', href: this.model.url(n)}, [
+      h.span({cls: classes.join(' '), 'aria-label': lnoStr}, []),
+    ]);
+  },
+  _render: function() {
+    var i;
+    var ctx_before = [], ctx_after = [];
+    var lno = this.model.get('lno');
+    var ctxBefore = this.model.get('context_before'), clip_before = this.model.get('clip_before');
+    var ctxAfter = this.model.get('context_after'), clip_after = this.model.get('clip_after');
+
+    var lines_to_display_before = Math.max(0, ctxBefore.length - (clip_before || 0));
+    for (i = 0; i < lines_to_display_before; i ++) {
+      ctx_before.unshift(
+        this._renderLno(lno - i - 1, false),
+        h.span([this.model.get('context_before')[i]]),
```

Comment:
IE doesn't like the extra ',' here and a few other places. Even with this fixed, IE still won't display these correctly, but at least it will load the js without error.